tests/run-make/compressed-debuginfo/rmake.rs

```diff
@@ -0,0 +1,28 @@
+// Checks the `debuginfo-compression` option.
+
+//@ only-linux
+//@ ignore-cross-compile
+
+use run_make_support::{assert_contains, cmd, path, rustc};
+
+fn check_compression(compression: &str, to_find: &str) {
+    let out = rustc()
+        .crate_name("foo")
+        .crate_type("lib")
+        .emit("obj")
+        .arg("-Cdebuginfo=full")
+        .arg(&format!("-Zdebuginfo-compression={compression}"))
+        .input("foo.rs")
+        .run();
+    let stderr = out.stderr_utf8();
+    if stderr.is_empty() {
+        cmd("readelf").arg("-t").arg(path("foo.o")).run().assert_stdout_contains(to_find);
+    } else {
+        assert_contains(&stderr, &format!("unknown debuginfo compression algorithm {compression}"));
+    }
```

Comment:
Discussion: I want to say that you *may* want to run each check inside [`run_in_tmpdir`](https://doc.rust-lang.org/nightly/nightly-rustc/run_make_support/fn.run_in_tmpdir.html) to prevent the output artifacts from potentially causing the checks to interfere with each other (at present, I think both check invocations work on the same `foo.o`). Don't think this is actually happening, but yeah.